docs/tutorials/tabular_prediction/tabular-custom-model.md

```diff
@@ -0,0 +1,374 @@
+- Image feature type support
+  - Refer to [ImagePredictorModel](../../_modules/autogluon/tabular/models/image_prediction/image_predictor.html#ImagePredictorModel)
+- Custom HPO logic
+  - Refer to LGBModel
+
+## Implementing a custom model
+
+Here we define the custom model we will be working with for the rest of the tutorial.
+
+The most important methods that must be implemented are `_fit` and `_preprocess`.
+
+Follow along with the code comments to better understand how the code works.
+
+```{.python .input}
+import numpy as np
+import pandas as pd
+
+from autogluon.core.models import AbstractModel
+from autogluon.features.generators import LabelEncoderFeatureGenerator
+
+class CustomRandomForestModel(AbstractModel):
+    def __init__(self, **kwargs):
+        # Simply pass along kwargs to parent, and init our internal `_feature_generator` variable to None
+        super().__init__(**kwargs)
+        self._feature_generator = None
+
+    # The `_preprocess` method takes the input data and transforms it to the internal representation usable by the model.
+    # `_preprocess` is called by `preprocess` and is used during model fit and model inference.
+    def _preprocess(self, X: pd.DataFrame, is_train=False, **kwargs) -> np.ndarray:
+        print(f'Entering the `_preprocess` method: {len(X)} rows of data (is_train={is_train})')
+        X = super()._preprocess(X, **kwargs)
+
+        if is_train:
+            # X will be the training data.
+            self._feature_generator = LabelEncoderFeatureGenerator(verbosity=0)
+            self._feature_generator.fit(X=X)
+        if self._feature_generator.features_in:
+            # This converts categorical features to numeric via stateful label encoding.
+            X = X.copy()
+            X[self._feature_generator.features_in] = self._feature_generator.transform(X=X)
+        # Add a fillna call to handle missing values.
+        # Some algorithms will be able to handle NaN values internally (LightGBM).
+        # In those cases, you can simply pass the NaN values into the inner model.
+        # Finally, convert to numpy for optimized memory usage and because sklearn RF works with raw numpy input.
+        return X.fillna(0).to_numpy(dtype=np.float32)
+
+    # The `_fit` method takes the input training data (and optionally the validation data) and trains the model.
+    def _fit(self,
+             X: pd.DataFrame,  # training data
+             y: pd.Series,  # training labels
+             # X_val=None,  # val data (unused in RF model)
+             # y_val=None,  # val labels (unused in RF model)
+             # time_limit=None,  # time limit in seconds (ignored in tutorial)
+             **kwargs):  # kwargs includes many other potential inputs, refer to AbstractModel documentation for details
+        print('Entering the `_fit` method')
+        from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
```

Comment:
demonstrating lazy import here might accelerate contributions

Reply:
added extra code comments for this